Source/WebCore/css/CSSSelectorList.cpp

```diff
@@ -115,12 +115,7 @@ String CSSSelectorList::selectorsText() const
 
 void CSSSelectorList::buildSelectorsText(StringBuilder& stringBuilder) const
 {
-    const CSSSelector* firstSubselector = first();
-    for (const CSSSelector* subSelector = firstSubselector; subSelector; subSelector = CSSSelectorList::next(subSelector)) {
-        if (subSelector != firstSubselector)
-            stringBuilder.append(", "_s);
-        stringBuilder.append(subSelector->selectorText());
-    }
+    stringBuilder.append(interleave(*this, [](auto& subSelector) { return subSelector.selectorText(); }, ", "_s));
```

Comment:
Sure wish this was `buildSelectorText(builder)` instead of `selectorText()`.

Reply:
Me too! I took a quick stab at it and it turned into a bit of an onion peel because of some weird-ish things selectorText() does right at the end:

```
        }
        return previousSelector->selectorText(separator, builder);
    } else if (auto separatorText = separatorTextForNestingRelative(); !separatorText.isNull()) {
        // We have a separator but no tag history which can happen with implicit relative nesting selector
        return separatorText + builder.toString();
    }

    return builder.toString();
```

Specifically that  "return separatorText + builder.toString();". I am sure there is a nice solution, but can be in its own PR.